stacks-signer/src/runloop.rs

```diff
@@ -937,21 +1071,60 @@ impl<C: Coordinator> SignerRunLoop<Vec<OperationResult>, RunLoopCommand> for Run
     }
 }
 
+fn check_nack_threshold(
```

Comment:
Add comments here and maybe rename it. It took me a minute to understand this was a function to "find the first of coordinator metadata where the nack threshold is reached"

Reply:
Thanks for pointing that out. Changing that function to return the coordinator_metadata that has the most broadcast _if_  it also meets the threshold.
I'm also adding a cleanup method that gets invoked whenever the current signer has a more updated view of the chain.